testcases/kernel/syscalls/mincore/mincore02.c

```diff
@@ -1,136 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
 /*
  * Copyright (c) International Business Machines  Corp., 2001
- *   Author: Rajeev Tiwari: rajeevti@in.ibm.com
+ * Author: Rajeev Tiwari: rajeevti@in.ibm.com
  * Copyright (c) 2004 Gernot Payer <gpayer@suse.de>
  * Copyright (c) 2013 Cyril Hrubis <chrubis@suse.cz>
- *
- * This program is free software;  you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY;  without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
- * the GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program;  if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ * Copyright (C) 2024 SUSE LLC Andrea Manzini <andrea.manzini@suse.com>
  */
 
-/*
+/*\
+ * [Description]
+ *
  * This test case provides a functional validation for mincore system call.
  * We mmap a file of known size (multiple of page size) and lock it in
  * memory. Then we obtain page location information via mincore and compare
  * the result with the expected value.
  */
 
-#include <sys/mman.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <sys/wait.h>
-#include <fcntl.h>
-#include <unistd.h>
-#include <signal.h>
-#include <errno.h>
-
-#include "test.h"
-#include "safe_macros.h"
+#include "tst_test.h"
 
-char *TCID = "mincore02";
-int TST_TOTAL = 1;
+#define NUM_PAGES 4
 
-static int fd = 0;
-static void *addr = NULL;
+static void *addr;
+static int size;
 static int page_size;
```

Comment:
`getpagesize()` result does not need to be stored as a value. Effectively we use always only `getpagesize() * NUM_PAGES`.